cedar-policy-core/src/authorizer/err.rs

```diff
@@ -22,11 +22,11 @@ use thiserror::Error;
 #[derive(Debug, PartialEq, Eq, Clone, Error)]
 pub enum AuthorizationError {
     /// Failed to eagerly evaluate entity attributes when initializing the `Evaluator`.
-    #[error("error occurred while evaluating entity attributes: {0}")]
+    #[error("Error occurred while evaluating entity attributes: {0}.")]
     AttributeEvaluationError(EvaluationError),
 
     /// An error occurred when evaluating a policy.
-    #[error("error occurred while evaluating policy `{}`: {}", &.id, &.error)]
+    #[error("Error occurred while evaluating policy '{}': {}.", &.id, &.error)]
```

Comment:
I recall now that the reason we decided not to use proper sentence capitalization/punctuation was to support "combining" error messages using colons. Because `&.error` is also an error message, the current change will result in messages like the following: 

```
Error occurred while evaluating policy 'policy2': Cannot access attribute of unspecified entity 'jobLevel'..
```

(Note the double period.) There are a couple ways to solve this: 1) continue using lowercase & no punctuation, 2) combine error messages using periods instead of colons.

I don't have strong feelings either way, but it seems like the precedent in Rust is the former (see, e.g. <https://rustc-dev-guide.rust-lang.org/diagnostics.html#diagnostic-structure>). Any other opinions?